HorizonEngine/src/HorizonEngine/App.cpp

```diff
@@ -14,8 +14,8 @@ namespace Hzn
 		registerComponents();
 		/*HZN_CORE_ASSERT(false, "application already initialized");*/
 		m_Instance = this;
-		m_Input = std::unique_ptr<Input>(Input::create());
-		m_Window = std::unique_ptr<Window>(Window::create(800, 600, "HorizonEngine"));
+		m_Input = Input::create();
+		m_Window = Window::create(800, 600, "HorizonEngine");
```

Comment:
Why did you cancel the use of unique_ptr here?

Reply:
The create methods return shared pointers so I receive the pointer to those instances as shared pointers.